core/crypto/eddsa/blinded25519.go

```diff
@@ -0,0 +1,169 @@
+// blinded25519.go - blinded EdDSA signatures
+// License: AGPL version 3
+// Copyright: 2021 - Anonymous contributor
+
+// This module implements the blinded signature scheme as used in
+// Tor for OnionV3, see A.2. Tor's key derivation scheme:
+// https://github.com/torproject/torspec/blob/main/rend-spec-v3.txt#L2268-L2326
+// Note that this implementation of the scheme uses different constants
+// for hashing the (factor)s and as thus is not wire format compatible.
+
+// Note that the reason that Tor doesn't need a separate Sign() function
+// for blinded private keys is that they use a different in-memory
+// representation of secret keys where the nonce seed is stored in expanded
+// form (instead of deriving it with sha512 for each signature)
+
+// It would be kind of useful to have
+// BlindedPrivateKey.FromBytes()
+// BlindedPrivateKey.ToBytes()
+// to enable passing around the derived keys, for instance to pass on
+// delegated privileges. Unfortunately checking the private scalars does
+// not seem trivial (since they are "malformed", ie no clamping), so it
+// seems like it could be a great footgun. Leaving it out for now, but
+// it's something to consider for the future I guess.
+
+// Note that use of Scalar.SetUniformBytes() is a bit unorthodox:
+// The API was designed to take 64 uniformly random bytes,
+// which are then reduced mod L. We "abuse" it by providing
+// our 32byte scalars as the 32 lower bytes and the upper 32 bytes all-zero
+// --- This way we get to initialize Scalar objects without the usual clamping.
+// There doesn't seem to be another API that lets us initialize 32-byte scalars
+// directly.
+
+// And lastly, note that it would be interesting to come up with a
+// blinding scheme that preserves the "validity" of the derived secret key
+// for use with x25519, perhaps a way to modify `factor` such that
+//            clamp(a)*clamp(factor)
+// ===  clamp(clamp(a)*clamp(factor))
+
+package eddsa
+
+import (
+	"crypto/ed25519"
+	"crypto/sha512"
+	"filippo.io/edwards25519"
+)
+
+type BlindedPrivateKey struct {
```

Comment:
https://golang.org/doc/effective_go#commentary
"Inside a package, any comment immediately preceding a top-level declaration serves as a doc comment for that declaration. Every exported (capitalized) name in a program should have a doc comment."